scripts/overton/BERTopic_script.py

```diff
@@ -0,0 +1,90 @@
+import pandas as pd
+import spacy
+from bertopic import BERTopic
+from sklearn.feature_extraction.text import CountVectorizer
+from sentence_transformers import SentenceTransformer
+import typer
+#FINAL_TABLE_PATH = 'C:/Users/Hubert/Documents/DarlingProject/Overton/EDA/final_table.parquet'
+#FOLDER_PATH = 'C:/Users/Hubert/Documents/DarlingProject/Overton/BERTopic'
+
+def topic_modelling(data, sentence_transformer, en, folder_path, nr_topics = 'auto', min_ngram = 1, max_ngram = 3,  top_n_words = 5, top_n_topics = 32, model_name = 'BERTopic_default'):
+    topic_model = BERTopic(embedding_model = sentence_transformer, nr_topics = nr_topics, top_n_words = top_n_words)
+    topics, probs = topic_model.fit_transform(data)
+
+    vectorizer_model = CountVectorizer(ngram_range = (min_ngram, max_ngram), stop_words = en.Defaults.stop_words)
+    topic_model.update_topics(data, topics, vectorizer_model = vectorizer_model)
+
+    model_path = folder_path + str('/models/') + model_name
+    topic_model.save(model_path)
+
+    info = topic_model.get_topic_info()
+    info_path = folder_path + str('/excel/') + model_name + str('_info.xlsx')
+    info.to_excel(info_path)
+
+    topics = topic_model.visualize_topics()
+    topics_path = folder_path + str('/plots/') + model_name + str('_topics.html')

```

Comment:
zamiast laczyc sciezki przez '+' polecam przez "os.path.join(). mozna dac wiele argumentow i jest odporne na roznice linux/windows

Reply:
To zrobiłem mix, bo część `+` skleja nazwę pliku

Reply:
To zrobiłem mix, bo część + skleja nazwę pliku